### PR TITLE
Add groupby service function

### PR DESCRIPTION
**Pull request by OscartGiles**
Adds two functions (commented on below) which provide cost summarised by service. 
- My formater went a bit mad so kinda hard to see the changes so i've added comment on them.

### CODE REVIEW

File: azure_usage/src_notebook/analysis.py [hunk 1]
```diff
@@ -215,25 +248,64 @@ def analyse_spnsr_usage(data_path, date_from, date_to, spnsr_budget):
...
+        [spnsr_ideal_avail_ym_df, spnsr_usage_ym_df.drop([CONST_COL_NAME_YM], axis=1)],
+        axis=1,
+    )
 
     return spnsr_remain, spnsr_usage, spnsr_usage_df, max_reg_cost_day
 
-def get_ideal_avail_spnsr_costs(sub_raw_data_df, spnsr_date_from, spnsr_date_to,
-    spnsr_budget, use_days = False):
+
+def analyse_spnsr_usage_by_service(data_path, date_from, date_to):
```
Comment (OscartGiles) [inline azure_usage/src_notebook/analysis.py]: New function

File: azure_usage/src_webapp/totals.py [hunk 2]
```diff
@@ -73,12 +86,46 @@ def group_year_month(raw_data):
...
+        raw_data_.groupby([CONST_COL_NAME_YM])[CONST_COL_NAME_COST]
+        .sum()
+        .reset_index()
+        .sort_values(by=[CONST_COL_NAME_YM], ascending=True)
+    )
 
     return raw_data_gr
 
+
+def group_year_month_service(raw_data):
```
Comment (OscartGiles) [inline azure_usage/src_webapp/totals.py]: new function

Comment (tomaslaz) [reply azure_usage/src_webapp/totals.py]: Could you please write unittests for the new function.

File: azure_usage/src_notebook/analysis.py (hunk 1 above)
Comment (tomaslaz) [reply azure_usage/src_notebook/analysis.py]: Could you please write unittests for the new function.

**Comment by tomaslaz** [comment]
Closing due to inactivity.

**Pull request closed without merging**